scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/conf/MetricsConfig.java

```diff
@@ -1,15 +1,42 @@
 package org.cloudfoundry.autoscaler.scheduler.conf;
 
-import io.prometheus.client.exporter.HTTPServer;
-import java.io.IOException;
+import io.prometheus.client.vertx.MetricsHandler;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.AuthProvider;
+import io.vertx.ext.auth.User;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BasicAuthHandler;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
 public class MetricsConfig {
 
-  @Bean(destroyMethod = "stop")
-  public HTTPServer metricsServer(MetricsConfiguration config) throws IOException {
-    return new HTTPServer(config.getPort());
+  @Bean(destroyMethod = "close")
+  Vertx metricsServer(MetricsConfiguration config) {
+    final Vertx vertx = Vertx.vertx();
+    final Router router = Router.router(vertx);
+    if (config.isBasicAuthEnabled()) {
+      router.route("/*").handler(BasicAuthHandler.create(getAuthProvider(config)));
+    }
+    router.route("/metrics").handler(new MetricsHandler());
+    vertx.createHttpServer().requestHandler(router).listen(config.getPort());
+    return vertx;
+  }
+
+  private AuthProvider getAuthProvider(MetricsConfiguration config) {
+    return (JsonObject authInfo, Handler<AsyncResult<User>> resultHandler) -> {
+      String password = authInfo.getString("password");
+      String username = authInfo.getString("username");
+      if (config.getUsername().equals(username) && config.getPassword().equals(password)) {
+        resultHandler.handle(Future.succeededFuture(null));
+      } else {
+        resultHandler.handle(Future.failedFuture("Not authorised"));
+      }
+    };
```

Comment:
Please check if this change can be replaced by using the new `Authenticator` support in [simpleclient_httpserver](https://github.com/prometheus/client_java/pull/682/files#diff-05c7094a6207ad292780607cc229e38691d89448909d71ab461b379ec0b1eb9dR246-R257)

The we would not need to add the VertX as dependency.

Reply:
We will take a look at it, thank you

Reply:
@silvestre good spot we have now reverted back to using the built in httpserver.